### DISCUSSION TITLE
How do I report test results to github classroom so they show up in the assignment overview page?

### DISCUSSION THREAD

**Discussion by ProfWentao** [community/community#144180] [GitHub Education]
### Hi everyone,

I am struggling to figure out how to get this done properly. Here's my setup of the template repo for a classroom template:

1. I have a set of input and expected output files for test cases in .github/classroom/ folder.
2. In .github/workflows/ci.yml file, I have the jobs to create the build and run the tests.
3. Question is how I can get the test results to show up on the assignment page.

My template repo is for C++ project. I set it up so students just need to implement the library for some data structure like linked list. I provide them with main.cpp for test runs that links with their library. I am all ears if there is a better approach to run test for c++ binaries.

Thanks!

**Comment by Dhananjay885339** [comment]
Wap in java code block

**Reply by ProfWentao** [reply]
can you provide some details about how your java code block works and where you place it? I wrote some custom code to call check-runs API, the information showed up on actions page but not propagated to the assignment view.
